lib/webbluetooth/bindings.js

```diff
@@ -0,0 +1,329 @@
+
+      }
+    }, function(err){
+      console.log('err scanning', err);
+    });
+
+  this.emit('scanStart');
+};
+
+NobleBindings.prototype.stopScanning = function() {
+  this._startScanCommand = null;
+
+  //TODO: need web api completed for this to work'=
+  this.emit('scanStop');
+};
+
+NobleBindings.prototype.connect = function(deviceUuid) {
+  var self = this;
+  console.log('connect', deviceUuid);
+  var peripheral = this._peripherals[deviceUuid];
+
+  // Attempts to connect to remote GATT Server.
+  peripheral.device.gatt.connect()
+    .then(function(gattServer){
+      peripheral.gattServer = gattServer;
+      console.log('peripheral connected', gattServer);
+      self.emit('connect', deviceUuid);
+    }, function(err){
+      console.log('err connecting', err);
+      self.emit('connect', deviceUuid, err);
+    });
+
+};
+
+NobleBindings.prototype.disconnect = function(deviceUuid) {
+  var peripheral = this._peripherals[deviceUuid];
+  if(peripheral.gattServer ){
+    peripheral.gattServer.disconnect();
+    this.emit('disconnect', deviceUuid);
+  }
+};
+
+NobleBindings.prototype.updateRssi = function(deviceUuid) {
+  var peripheral = this._peripherals[deviceUuid];
+
+  //TODO: need web api completed for this to work
+  this.emit('rssiUpdate', deviceUuid, rssi);
+};
+
+NobleBindings.prototype.discoverServices = function(deviceUuid, uuids) {
+  var peripheral = this._peripherals[deviceUuid];
+
+  //TODO: need web api completed for this to work
+  if(peripheral){
+    this.emit('servicesDiscover', deviceUuid, peripheral.device.uuids);
+  }
+
+};
+
+NobleBindings.prototype.discoverIncludedServices = function(deviceUuid, serviceUuid, serviceUuids) {
+  var peripheral = this._peripherals[deviceUuid];
+
+  //TODO impelment when web API has functionatility then emit response
+  //this.emit('includedServicesDiscover', deviceUuid, serviceUuid, includedServiceUuids);
+};
+
+NobleBindings.prototype.discoverCharacteristics = function(deviceUuid, serviceUuid, characteristicUuids) {
+  var self = this;
+  var peripheral = self._peripherals[deviceUuid];
+
+  if(peripheral){
+    
+    self._getPrimaryService(peripheral, serviceUuid)
+      .then(function(service){
+        return service.getCharacteristics();
+      })
+      .then(function(characteristics) {
+        var discoveredCharateristcs = characteristics.map(function(char){
+          var charInfo = {uuid: stripDashes(char.uuid), properties: []};
+          
+          if(char.properties.writeWithoutResponse){
+            charInfo.properties.push('writeWithoutResponse');
+          }
+          
+          if(char.properties.write){
+            charInfo.properties.push('write');
+          }
+          
+          if(char.properties.read){
+            charInfo.properties.push('read');
+          }
+          
+          if(char.properties.notify){
+            charInfo.properties.push('notify');
+          }
+          
+          return charInfo;
```

Comment:
For info, here are all characteristic properties: https://googlechrome.github.io/samples/web-bluetooth/characteristic-properties.html